test/anchor_sup_tests.erl

```diff
@@ -0,0 +1,47 @@
+-module(anchor_sup_tests).
+-include_lib("anchor/include/anchor.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+anchor_sup_test_() ->
+    {foreach,
+     fun setup/0,
+     fun teardown/1,
+     [fun init_creates_all_pools/0,
+      fun init_uses_config_from_env/0]}.
+
+setup() ->
+    meck:new(shackle_pool),
+    meck:expect(shackle_pool, start, fun(_, _, _, _) -> ok end),
+    ok.
+
+teardown(_) ->
+    meck:unload(shackle_pool).
+
+init_creates_all_pools() ->
+    os:putenv("pools", "[{foo, []}, {bar, []}]"),
```

Comment:
Are you sure you don't mean `application:set_env`? That sets an OS environment variable, not an Erlang application one.

Reply:
Good catch! Damn this test shouldn't pass... I might be misusing `meck:num_called/3`...